nlp_apps.ipynb

```diff
@@ -860,6 +840,141 @@
+      "Paper No. 53: Hamilton: -0.334527 Madison: -0.329361 Jay: -0.336112\n",
+      "Paper No. 54: Hamilton: -0.333739 Madison: -0.328651 Jay: -0.337610\n",
+      "Paper No. 55: Hamilton: -0.334172 Madison: -0.328874 Jay: -0.336955\n",
+      "Paper No. 56: Hamilton: -0.334131 Madison: -0.328213 Jay: -0.337655\n",
+      "Paper No. 57: Hamilton: -0.333538 Madison: -0.329196 Jay: -0.337267\n",
+      "Paper No. 58: Hamilton: -0.333574 Madison: -0.328211 Jay: -0.338214\n",
+      "Paper No. 18: Hamilton: -0.340253 Madison: -0.331734 Jay: -0.328013\n",
+      "Paper No. 19: Hamilton: -0.338314 Madison: -0.332076 Jay: -0.329610\n",
+      "Paper No. 20: Hamilton: -0.337462 Madison: -0.332191 Jay: -0.330347\n",
+      "Paper No. 64: Hamilton: -0.334279 Madison: -0.333268 Jay: -0.332454\n"
+     ]
+    }
+   ],
+   "source": [
+    "disputed = list(range(49, 58+1)) + [18, 19, 20, 64]\n",
+    "jay, madison, hamilton = [], [], []\n",
+    "for i, p in enumerate(stemmed_papers):\n",
+    "    if i in disputed or i == 0:\n",
+    "        continue\n",
+    "    \n",
+    "    if 'jay' in p:\n",
+    "        jay.append(p)\n",
+    "    elif 'madison' in p:\n",
+    "        madison.append(p)\n",
+    "    else:\n",
+    "        hamilton.append(p)\n",
+    "\n",
+    "hamilton = ''.join(hamilton)\n",
+    "hamilton_words = words(hamilton)\n",
+    "P_hamilton = UnigramWordModel(hamilton_words, default=1)\n",
+    "\n",
+    "madison = ''.join(madison)\n",
+    "madison_words = words(madison)\n",
+    "P_madison = UnigramWordModel(madison_words, default=1)\n",
+    "\n",
+    "jay = ''.join(jay)\n",
+    "jay_words = words(jay)\n",
+    "P_jay = UnigramWordModel(jay_words, default=1)\n",
+    "\n",
+    "dist = {('Madison', 1): P_madison, ('Hamilton', 1): P_hamilton, ('Jay', 1): P_jay}\n",
+    "nBS = NaiveBayesLearner(dist)\n",
+    "nBSL = NaiveBayesLearnerLog(dist)\n",
+    "\n",
+    "print('\\nModel performance after using a simple stemmer')\n",
+    "print('\\nStraightforward Naive Bayes Learner\\n')\n",
+    "for d in disputed:\n",
+    "    probs = recognize(papers[d], nBS)\n",
+    "    results = ['{}: {:.4f}'.format(name, probs[(name, 1)]) for name in 'Hamilton Madison Jay'.split()]\n",
+    "    print('Paper No. {}: {}'.format(d, ' '.join(results)))\n",
+    "\n",
+    "print('\\nLogarithmic Naive Bayes Learner\\n')\n",
+    "for d in disputed:\n",
+    "    probs = recognize(papers[d], nBSL)\n",
+    "    results = ['{}: {:.6f}'.format(name, probs[(name, 1)]) for name in 'Hamilton Madison Jay'.split()]\n",
+    "    print('Paper No. {}: {}'.format(d, ' '.join(results)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The stemmer didn't seem to help much here. But the model classifying Madison's papers as Jay's after stemming the words may mean that they used the words having same root words. The classifier could perform well if we had more instances of their work."
```

Comment:
This looks better, nice work! I noticed a little issue with the last paragraph though. You write this:

"stemming the words may mean that they used the words having same root words."

First of all, you use the word 'words' three times in quick succession. This should generally be avoided. Second, the sentence needs to be made a bit clearer. Right now it is hard to understand what you mean.